RFCs/0033-graph-query.md

```diff
@@ -0,0 +1,491 @@
+
+<selector> ::=
+      [ 'ANY' | 'ALL' ] 'SHORTEST'
+    | 'ANY' <<integer>>?
+    | 'SHORTEST' <<integer>> 'GROUP'?
+
+<restrictor> ::= 
+    'TRAIL'  |  'ACYCLIC'  |  'SIMPLE'
+
+<node_pattern>  ::=  '(' <node_labeled> <<where_clause>>? ')'
+      
+<node_labeled> ::= 
+      <node_variable> ':' <label_descr>
+    | <node_variable> 
+    |                 ':' <label_descr>
+      
+<node_variable> ::=  <<identifier>>
+
+<label_descr> ::=              //Simplified. GPML provides %, !, |, &, grouping. 
+      <label_name>
+      
+<label_name> ::=  <<identifier>>
+
+<edge_pattern> ::=  <edge_spec> <quantifier>?
+
+<group_pattern> ::=
+      '(' <path_pattern> <<where_clause>>? ')' <quantifier>?
+    | '[' <path_pattern> <<where_clause>>? ']' <quantifier>?
+
+<quantifier> ::=
+      '+'  |  '*'
+    | '{' <<integer>> ',' <<integer>>? '}'
+    
+<edge_spec> ::= 
+      '<-'  |  '<-' '[' <edge_filtered> ']' '-'  
+    | '~'   |  '~'  '[' <edge_filtered> ']' '~' 
+    | '->'  |  '-'  '[' <edge_filtered> ']' '->' 
+    | '<~'  |  '<~' '[' <edge_filtered> ']' '~' 
+    | '~>'  |  '~'  '[' <edge_filtered> ']' '~>' 
+    | '<->' |  '<-' '[' <edge_filtered> ']' '->' 
+    | '-'   |  '-'  '[' <edge_filtered> ']' '-' 
+    
+<edge_filtered> ::=  <edge_labeled> <<where_clause>>?
+    
+<edge_labeled> ::= 
+      <edge_variable> ':' <label_descr>
+    | <edge_variable> 
+    |                 ':' <label_descr>
+    
+<edge_variable> ::=  <<identifier>>
+```
+
+It is worthy of a note that `<<where_clause>>`, referenced in the above grammar 
+in node, edge, and group patterns, is, syntactically, the `WHERE` clause of host PartiQL,
+where it is defined as 
+```EBNF
+<where_clause> ::=  'WHERE' <expr_query>
+```
+where the expression `<expr_query>` is meant to evaluate to a boolean. 
+This expession, however, can contain, as references, variables introduced as 
+`<node_variable>`, `<edge_variable>`, or `<path_variable>` 
+elsewhere in the pattern.
+
+
+### On WHERE within MATCH
+
+In a few places, [^gpml-paper] mentions `WHERE` clause that is not a part of any 
+node, edge, or path fragment of a pattern, 
+but rather applies to the entirety of the pattern in a graph match expression. 
+This WHERE is sometimes referred to as a "postfilter" of a pattern.
+From the limited discussion in the paper and the lack of examples on the matter, 
+it is not fully clear whether such `WHERE` is meant to be the `WHERE` of the "host" 
+select-from-where query (in SQL or PartiQL) or it is a new kind of `WHERE` 
+that is coupled with `MATCH`. 
+
+In the case of the latter, the syntax of graph match expression would have to accommodate 
+this possibility with a grammar production like
+```EBNF
+<expr_query> ::=
+  // ...
+   |  '(' <expr_query> 'MATCH' <graph_pattern> [ 'WHERE' <expr_query> ]?  ')'
+```
+
+Currently, this is not being proposed for the reasons of economy of design -- 
+until and unless it becomes certain that the "host" `WHERE` 
+cannot support necessary use cases.
```

Comment:
Structurally, I would put this in the open questions section as we are not planning on implementing it, but it is open as to what will actually be defined here.